### ISSUE TITLE
Notifications: Comments from the Reader or the notifications dropdown don't trigger an email notification

### ISSUE THREAD

**Issue by designsimply** [Automattic/wp-calypso#976]
Raised by @rachelmcr 

A site's <a href="https://wordpress.com/settings/discussion/">discussion settings</a> have the option to email the owner when certain actions are taken on the blog:

![screen shot 2015-09-15 at 11 05 00 am](https://cloud.githubusercontent.com/assets/8658164/9873745/a9ad8088-5b99-11e5-8705-0dfe39a52f38.png)

When a comment is held for moderation, the email notification about that comment is sent as expected. 

However, if a comment is posted immediately (without moderation) from the Reader or from the notifications dropdown, it doesn't trigger an email notification even if that first setting ("Anyone posts a comment") is checked.

If the comment is posted immediately from the front of the site or via email, it does trigger an email notification as expected.

Steps to reproduce:

1. On a test blog, open the discussion settings and confirm that "Anyone posts a comment" is checked, and "Comment must be manually approved" and "Comment author must have a previously approved comment" are not checked. (Comments will be published immediately on the blog, and the blog owner should receive email notifications about each comment.)
1. Using a test account that is not a user on the blog, follow the blog (and opt to receive updates via email).
1. From the account that owns the blog, publish a new post.
1. From the test account, leave a comment by replying to the email notification about the new post. Result: The blog owner receives an email notification.
1. From the test account, visit the front of the blog and leave a comment on the post. Result: The blog owner receives an email notification.
1. From the Reader, open the post and leave a comment. Result: The blog owner does *not* receive an email notification.
1. From the account that owns the blog, leave a comment as a reply to one of the test account's comments. This triggers a notification for the test account.
1. From the test account, open the notifications dropdown and reply to the comment notification. Result: The blog owner does *not* receive an email notification.

Tickets: 662069-f, 662637-f, 2352165-t, 664417-f

Note from @blowery 

> Both the Reader and Notifications run through the REST API, wonder if the missing notification is hiding in there somewhere...
> 
> Thanks for the report and detailed steps @rachelmcr !

**Comment by blowery** [comment]
Verified that we're just using the REST API and creating a comment the normal way. Dropping the Reader label for now, because I don't think it's anything in the Reader.

@dmsnell might know more?

**Comment by dmsnell** [comment]
calling @beaulebens because I tried looking into this but am lost in the email notifications code

**Comment by beaulebens** [comment]
I _think_ those specific emails are actually for the author of the post (or a moderator), not necessarily for the owner of a site. They are core WordPress (not our "social notifications" layer), so should just be getting triggered via core hooks etc. It looks like we use `wp_new_comment()` in the API, which fires the `comment_post` hook, which appears to be where core notifications are hooked to: https://core.trac.wordpress.org/browser/trunk/src/wp-includes/default-filters.php#L355

Which is to say -- I'm not sure why the emails are not being sent either :)

**Comment by hoverduck** [comment]
21Apr16 Bug Scrub

Confirmed this is still an issue.  I followed a post as a secondary user, and confirmed I would receive e-mails on new comments posted via the form on the post.  I did not receive e-mails on new comments posted via the notifications' Reply field on the web (OSX Chrome)

However, I did receive an e-mail on a comment posted as a reply from the notifications field on the Android app.  I did not test iOS.
